pages/new/index.tsx

```diff
@@ -61,12 +67,26 @@ function HomePage({ policyType, csrfToken, lang }: HomePageProps) {
 }
 
 export const getServerSideProps: GetServerSideProps = async (context) => {
-  const { csrfToken } = getCsrfToken(context.req, context.res);
+  const tokenFromRequestCookie = parseTokenFromPostRequestCookies(context.req);
   const { query } = context;
-  context.res.setHeader(
-    "Set-Cookie",
-    `nyplUserRegistered=false; Max-Age=-1; path=/; domain=${cookieDomain};`
-  );
+  let newTokenCookie;
+  let csrfToken = tokenFromRequestCookie.value;
+  if (!csrfToken) {
+    csrfToken = generateNewToken();
+    const newTokenCookieString = generateNewCookieTokenAndHash(csrfToken);
+    newTokenCookie = serialize(
```

Comment:
If you do remove this export from cookie directly from this file and move it to `CookieUtils`, you can create a new "serialize" function that only needs the `newTokenCookieString` argument since `metadata` is already in the same file.

Reply:
brilliant! thanks. i knew there was something fishy going on there